SoftLayer/managers/block.py

```diff
@@ -184,6 +190,44 @@ def deauthorize_host_to_volume(self, volume_id,
         return self.client.call('Network_Storage', 'removeAccessFromHostList',
                                 host_templates, id=volume_id, **kwargs)
 
+    def order_replicant_volume(self, volume_id, snapshot_schedule,
+                               location, tier, os_type):
+        """Places an order for a replicant block volume.
+
+        :param volume_id: The ID of the primary volume to be replicated
+        :param snapshot_schedule: The primary volume's snapshot
+                                  schedule to use for replication
+        :param location: The location for the ordered replicant volume
+        :param tier: The tier (IOPS per GB) of the primary volume
+        :param os_type: The OS type of the primary volume
+        :return: Returns a SoftLayer_Container_Product_Order_Receipt
+        """
+
+        block_mask = 'billingItem[activeChildren],storageTierLevel,'\
+                     'osType,snapshotCapacityGb,schedules,'\
+                     'hourlySchedule,dailySchedule,weeklySchedule'
+        block_volume = self.get_block_volume_details(volume_id,
+                                                     mask=block_mask)
+
+        if os_type is None:
+            if 'osType' in block_volume\
```

Comment:
There is a utility function in SoftLayer.utils for looking up things in a nested dict much easier. For example:
```
        if isinstance(utils.lookup(block_volume, 'osType', 'keyName'), str):
```

Also, all this logic is to make os_type optional. Typically that is done by making os_type a keyword argument with a default value. In this case, the appropriate default is probably `None`.